scripts/importer/data_importer/reference_set_importer.py

```diff
@@ -0,0 +1,463 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import gzip
+import time
+import shutil
+import logging
+import zipfile
+import db
+from peewee import IntegrityError, fn
+from .data_importer import DataImporter
+
+class ReferenceSetImporter( DataImporter ):
+
+    GENCODE = "ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_{a.gencode_version}/gencode.v{a.gencode_version}.annotation.gtf.gz"
+    DBNSFP = "ftp://dbnsfp:dbnsfp@dbnsfp.softgenetics.com/dbNSFPv{a.dbnsfp_version}.zip"
+    ENSEMBL = ("ensembldb.ensembl.org", "anonymous", "")
+
+    def __init__(self, settings):
+        super().__init__(settings)
+
+        # counters for statistics and progress
+        self.numbers = {'genes':None,
+                        'transcripts':None,
+                        'features':None}
+        self.counters = {'genes':0,
+                        'transcripts':0,
+                        'features':0}
+
+        # dictionaries to hold data while processing
+        self.genes = []
+        self.transcripts = []
+        self.features = []
+
+        # file handlers for later
+        self.gencode = None
+        self.dbnsfp  = None
+        self.omim    = None
+        self.ensembl = None
+
+    def _insert_features(self):
+        logging.info("Inserting features into database")
+        start = time.time()
+        self._print_progress_bar()
+        last_progress = 0
+        batch = []
+        with db.database.atomic():
+            for i, feature in enumerate(self.features):
+                batch += [{'gene':self.gene_db_ids[feature['gene_id']],
+                        'transcript':self.transcript_db_ids[feature['transcript_id']],
+                        'chrom':feature['chrom'],
+                        'start':feature['start'],
+                        'stop':feature['stop'],
+                        'strand':feature['strand'],
+                        'feature_type':feature['feature_type']}]
+
+                if len(batch) % self.batch_size == 0:
```

Comment:
In all other places this test is done with `len(batch) >= self.batch_size` instead. Any reason for doing it this way here? This one will trigger in case batch is `empty`.

Reply:
Check is now using >=
Also fixed a few other dubious things in the area, e.g. if len(batch):